edtLHD/src/main/java/fr/univtln/lhd/facade/Schedule.java

```diff
@@ -151,6 +212,48 @@ public static List<Slot> getSchedule(Group group, Interval timerange) {
         }
         return schedule.filterEverySlotOutsideTimerange(allSlotOfGroup, timerange);
     }
+    /**
+     * Round to certain number of decimals
+     * Author:jav-rock
+     * @param d
+     * @param decimalPlace
+     * @return
+     */
+    public static float round(float d, int decimalPlace) {
+        BigDecimal bd = new BigDecimal(Float.toString(d));
+        bd = bd.setScale(decimalPlace, RoundingMode.FLOOR);
+        return bd.floatValue();
+    }
+
+    /**
+     * Take a group and a slot and return a percentage of
+     * the number of slot passed on the number of slot in the future
+     * for the group
+     * it is a rough estimate and doesn't take account the difference of length between slot
+     *
+     * @param group
+     * @param slot
+     * @return
+     */
+    public static float getPercentageOf(Group group, Slot slot) {
+        int total = 1;
+        int passed = 0;
+        List<Slot> listOfSlot;
+        SlotDAO dao = SlotDAO.getInstance();
+        try {
+            listOfSlot = dao.getSlotOfGroup(group);
+        } catch (SQLException e) {
+            return -1;
+        }
+        for (Slot xslot :
+                listOfSlot) {
+            if (xslot.getTimeRange().isBefore(slot.getTimeRange())) {
+                passed += 1;
+            }
+            total += 1;
```

Comment:
`total += 1`
Can be removed entirely, use 
`total = slot.getSubject().getHourCountMax()`

Doing those two changes should give you the estimate percentage of passed subject over subject max hours